### DISCUSSION TITLE
Import .svg and resize .svg in styled components

### DISCUSSION THREAD

**Discussion by mikekubn** [vercel/next.js#34915] [Help] [answered]
Hi, I am currently trying import `.svg` into next.js project but I failed.

- I tried import `.svg` same way as in react project, that I created `typing.d.ts` and import svg like component. But it doesn't work.

```.ts
declare module '*.svg' {
  import React from 'react';

  const src: string;

  export const ReactComponent: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
  export default src;
}

```

- When I started looking how can I import `.svg` in next.js project, so I found few examples:

[svg-components](https://github.com/vercel/next.js/tree/canary/examples/svg-components)
[image-component](https://github.com/vercel/next.js/tree/canary/examples/image-component)

- But that didn't work in the case for me, if I needed resize width or height of `.svg`.
- In next.js documentation is describe how use `next/image` - [image](https://nextjs.org/docs/api-reference/next/image#required-props) and that svg is dangerously [svg](https://nextjs.org/docs/api-reference/next/image#dangerously-allow-svg)

**Example**
- I have created a small sample of my problem: [next.js-svg-import](https://stackblitz.com/edit/nextjs-bubipe?file=package.json)

version:
next.js - 12.1
styled-components - 5.3.3
Has anyone had the same problem? Or did anyone manage to solve it? Thank you

**Comment by mikekubn** [comment, marked as answer]
### Maybe a solution.

- Default option in next.js for removeViewBox is true. If you change in .svg height or width the view-box from .svg will be rest to default value.

- If you use in your next.config.js webpack and set removeViewBox to false .svg resize start work.

- A working example: [stackblitz-resize-svg-next.js](https://stackblitz.com/edit/nextjs-mbppty?file=next.config.js)

```.config.js
 webpack(config) {
    config.module.rules.push({
      loader: '@svgr/webpack',
      options: {
        prettier: false,
        svgo: true,
        svgoConfig: {
          plugins: [
            {
              name: 'preset-default',
              params: {
                overrides: { removeViewBox: false },
              },
            },
          ],
        },
        titleProp: true,
      },
      test: /\.svg$/,
    });

    return config;
  }
...
```

- **If you have in .babelrc set "inline-react-svg" this solution will not work.**

**Reply by TheJazzDev** [reply]
> ### Maybe a solution.
> * Default option in next.js for removeViewBox is true. If you change in .svg height or width the view-box from .svg will be rest to default value.
> * If you use in your next.config.js webpack and set removeViewBox to false .svg resize start work.
> * A working example: [stackblitz-resize-svg-next.js](https://stackblitz.com/edit/nextjs-mbppty?file=next.config.js)
> 
> ```js
>  webpack(config) {
>     config.module.rules.push({
>       loader: '@svgr/webpack',
>       options: {
>         prettier: false,
>         svgo: true,
>         svgoConfig: {
>           plugins: [
>             {
>               name: 'preset-default',
>               params: {
>                 overrides: { removeViewBox: false },
>               },
>             },
>           ],
>         },
>         titleProp: true,
>       },
>       test: /\.svg$/,
>     });
> 
>     return config;
>   }
> ...
> ```
> 
> * **If you have in .babelrc set "inline-react-svg" this solution will not work.**

Thanks a lot... This works for me...

**Reply by Maxssobolev** [reply]
works as expected!

**Reply by Artz8489** [reply]
Worked for me thanks
